src/apiClient/approvals.ts

```diff
@@ -0,0 +1,90 @@
+import { toIsoDate } from "lib/fmtDate";
+import { request } from "./client";
+
+export type Approval = {
+  id: number;
+  startDate: string;
+  endDate: string;
+  note: string;
+  approvedBy: Person;
+  renter: Person;
+  items: ApprovalItem[];
+};
+
+interface GearItem {
+  id: string;
+  type: GearType;
+}
+
+export enum ApprovalItemType {
+  gearType = "gearType",
+  specificItem = "specificItem",
+}
+
+type GenericGearTypeApproval<T> = {
+  type: ApprovalItemType.gearType;
+  item: {
+    quantity: number;
+    gearType: T;
+  };
+};
+
+type GenericSpecificItemApproval<T> = {
+  type: ApprovalItemType.specificItem;
+  item: {
+    gearItem: T;
+  };
+};
+
+export type ApprovalItem =
+  | GenericGearTypeApproval<GearType>
+  | GenericSpecificItemApproval<GearItem>;
+
+export type PartialApprovalItem =
+  | GenericGearTypeApproval<number | undefined>
+  | GenericSpecificItemApproval<string | undefined>;
+
+export type ApprovalItemToCreate =
+  | GenericGearTypeApproval<number>
+  | GenericSpecificItemApproval<string>;
+
+interface Person {
```

Comment:
There's an interface for person embedded in the `GearRental` type? This type might benefit from being inside the `people.ts` file regardless? I'm not sure what the TypeScript convention is here for duplicating types everywhere.

Mild ditto for the `GearType` - there's a `GearType` type within `gear.ts` that includes a `defaultDeposit` too, which I know are different but seems like there's possibility for reuse.

Reply:
Good point, I refactored the types